File: data-label-api/src/main/java/com/btd/service/AnnotationService.java
package com.btd.service;

import java.io.IOException;
import java.util.Set;

import javax.inject.Inject;
import javax.ws.rs.ForbiddenException;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.btd.model.Annotation;
import com.btd.model.Image;
import com.btd.model.User;
import com.btd.repository.AnnotationRepository;
import com.btd.repository.UserRepository;

@Service
@Transactional
public class AnnotationService {

	@Autowired
	private UserRepository userRepository;
	@Autowired
	private AnnotationRepository annotationRepository;
	@Inject
	private ImageService imageService;

	public Set<Annotation> mark(Set<Annotation> annotations, Long imageId, String token) throws IOException {

		User user = userRepository.findByToken(token);

		if (user == null) {
			throw new ForbiddenException("Access denied");
		}

		Image image = imageService.get(imageId);

		annotations.forEach(annotation -> {
			annotation.setImage(image);
			annotationRepository.save(annotation);
		});

		return annotations;
	}
}

File: data-label-web/src/app/annotation.service.ts
import { Injectable } from '@angular/core';
import {UserService} from './user.service';
import {HttpClient} from '@angular/common/http';
import {Image} from './model/image';
import {Annotation} from './model/Annotation';

@Injectable()
export class AnnotationService {

  constructor(private _http: HttpClient, private _userService: UserService) {
  }

  save(annotations: Annotation[], userId: number, imageId: number,  token: string) {
    return this._http.post(this._userService.user_url + '/' + userId + '/images/' + imageId + '/annotations?token=' + token, annotations);
  }

}
